wpilib/src/pneumatics.rs

```diff
@@ -218,101 +238,87 @@ impl DoubleSolenoid {
-        self.set_closed_loop_control(false);
+    /// Forcibly stop the compressor.
+    pub fn stop(&mut self) -> HalResult<()> {
+        self.set_closed_loop_control(false)
     }
 
     pub fn enabled(&self) -> bool {
-        maybe_hal_call!(HAL_GetCompressor(self.compressor_handle)).ok() != 0
+        maybe_hal_call!(HAL_GetCompressor(self.handle)).ok() != 0
     }
 
     pub fn pressure_switch_value(&self) -> bool {
-        maybe_hal_call!(HAL_GetCompressorPressureSwitch(self.compressor_handle)).ok() != 0
+        maybe_hal_call!(HAL_GetCompressorPressureSwitch(self.handle)).ok() != 0
     }
 
-    pub fn compressor_current(&self) -> f64 {
-        maybe_hal_call!(HAL_GetCompressorCurrent(self.compressor_handle)).ok()
+    pub fn current(&self) -> f64 {
+        maybe_hal_call!(HAL_GetCompressorCurrent(self.handle)).ok()
     }
 
     pub fn closed_loop_control(&self) -> bool {
-        maybe_hal_call!(HAL_GetCompressorClosedLoopControl(self.compressor_handle)).ok() != 0
+        maybe_hal_call!(HAL_GetCompressorClosedLoopControl(self.handle)).ok() != 0
     }
 
-    pub fn compressor_current_too_high_fault(&self) -> bool {
-        maybe_hal_call!(HAL_GetCompressorCurrentTooHighStickyFault(
-            self.compressor_handle
-        ))
-        .ok()
-            != 0
+    pub fn current_too_high_fault(&self) -> bool {
+        maybe_hal_call!(HAL_GetCompressorCurrentTooHighFault(self.handle)).ok() != 0
     }
 
-    pub fn compressor_current_too_high_sticky_fault(&self) -> bool {
-        maybe_hal_call!(HAL_GetCompressorCurrentTooHighStickyFault(
-            self.compressor_handle
-        ))
-        .ok()
-            != 0
-    }
-
-    pub fn compressor_shorted_sticky_fault(&self) -> bool {
-        maybe_hal_call!(HAL_GetCompressorShortedStickyFault(self.compressor_handle)).ok() != 0
+    pub fn current_too_high_sticky_fault(&self) -> bool {
+        maybe_hal_call!(HAL_GetCompressorCurrentTooHighStickyFault(self.handle)).ok() != 0
     }
 
-    pub fn compressor_shorted_fault(&self) -> bool {
-        maybe_hal_call!(HAL_GetCompressorShortedFault(self.compressor_handle)).ok() != 0
+    pub fn shorted_sticky_fault(&self) -> bool {
+        maybe_hal_call!(HAL_GetCompressorShortedStickyFault(self.handle)).ok() != 0
     }
 
-    pub fn compressor_not_connected_sticky_fault(&self) -> bool {
-        maybe_hal_call!(HAL_GetCompressorNotConnectedStickyFault(
-            self.compressor_handle
-        ))
-        .ok()
-            != 0
+    pub fn shorted_fault(&self) -> bool {
+        maybe_hal_call!(HAL_GetCompressorShortedFault(self.handle)).ok() != 0
     }
 
-    pub fn compressor_not_connected_fault(&self) -> bool {
-        maybe_hal_call!(HAL_GetCompressorNotConnectedFault(self.compressor_handle)).ok() != 0
+    pub fn not_connected_sticky_fault(&self) -> bool {
+        maybe_hal_call!(HAL_GetCompressorNotConnectedStickyFault(self.handle)).ok() != 0
     }
 
-    pub fn clear_all_pcm_sticky_faults(&self) {
```

Comment:
should we not keep this? http://first.wpi.edu/FRC/roborio/beta/docs/java/edu/wpi/first/wpilibj/Compressor.html#clearAllPCMStickyFaults--

Reply:
This is on PneumaticsControlModule where it logically belongs.

Reply:
Though I agree, this does bring up a broader issue about whether we prefer compatibility with WPILib or what logically makes sense. Either way, would it make sense to have a module getter on `Compressor` like `Solenoid` has. That way the API is more uniform and clearing faults from a compressor is still possible.

Reply:
`Solenoid` returns an immutable reference in this current state, so there's no clean way to clear the PCM's sticky faults given a `Solenoid` either. (I wouldn't call `.clone()` clean.)

Reply:
That makes sense, but this points to a larger problem. Returning an immutable reference to a PCM is useless, as _all_ methods on PCMs take `self` or `&mut self`. One of those facts should change before merge.

Reply:
PCM is Copy, so the methods that take `self` are fine.

Reply:
I thought I tested that, but I went back to test it again, and I was wrong. The methods that take `self` are fine. My concern was that people could have solenoids they couldn't clear faults for, but now I see that that isn't an issue, as any PCM that isn't `::default()` needs to be constructed in code before people can create solenoids on it.